Re-land [TestWebKitAPI] Fix the public iOS 26 build

Smackteo opened the pull request:
#### 0f6a1d58124bf8c91ac8b8270a1204d29336dee3
<pre>
Re-land [TestWebKitAPI] Fix the public iOS 26 build
<a href="https://bugs.webkit.org/show_bug.cgi?id=300036">https://bugs.webkit.org/show_bug.cgi?id=300036</a>

Reviewed by Abrar Rahman Protyasha.

Update UITextInput protocol in SPI headers (this protocol is actually
API, but WKWebView still uses a pre-API internal protocol). This needs
to be visible from Swift, so add it to the bridging header. Unlike our
frameworks, the test bundle is not itself a module, so we do not need to
modularize UIKitSPIForTesting.h and friends to be able to use it from
swift code.

Along the way, fix a couple of public build issues, noted below.

Test: Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h

Test: Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h

* Configurations/CommonBase.xcconfig: Change the definition of
  SDKDB_HAS_148943382 to be internal only. When we generate partial
  SDKDBs from the public SDK, they do not exhibit the same bug. This
  change reverts an earlier attempt to work around the bug in
  300877@main.

* Source/WebCore/Configurations/AllowedSPI-legacy.toml: Revert an
  earlier attempt to fix the above problem (300877@main).

* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h:
* Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h:

* WebKitLibraries/SDKs/iphoneos26.0-additions.sdk/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore.tbd:
  Removed. The public build gets UIKit symbols from UIKit.tbd directly,
  not by binding to UIKitCore. This sidesteps an issue where WebKitSwift
  binds to different UIKit symbols in debug and release configurations,
  which confuses extract-tbds-from-internal-sdk.
* WebKitLibraries/SDKs/iphonesimulator26.0-additions.sdk/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore.tbd: Removed.

Originally-landed-as: 300969@main (77271dc9c550). <a href="https://bugs.webkit.org/show_bug.cgi?id=300036">https://bugs.webkit.org/show_bug.cgi?id=300036</a>
Canonical link: <a href="https://commits.webkit.org/301106@main">https://commits.webkit.org/301106@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fe4b9bfde8099d6893ae1fc29a28a9d7f6a24333

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124932 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44601 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35338 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131782 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76815 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/974f2c6e-8442-4306-9d02-19ee7f76c583) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126809 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45294 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53168 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95080 "") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63077 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b114ba43-e539-433b-9e72-4035bc713107) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127886 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36145 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111732 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75629 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6f5435b4-9be3-4245-9def-33a69af54d54) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/35080 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29885 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75259 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105907 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30116 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134453 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51763 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39565 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103560 "Failed to checkout and rebase branch from PR 51895") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52184 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107945 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103334 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26313 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48676 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26968 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48782 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51642 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57444 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51024 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54379 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52717 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

bls1999 reviewed (approved):
unofficial rs=me, assuming this works with public/internal builds of iOS 18/26

emw-apple committed:
Re-land [TestWebKitAPI] Fix the public iOS 26 build
https://bugs.webkit.org/show_bug.cgi?id=300036

Reviewed by Abrar Rahman Protyasha.

Update UITextInput protocol in SPI headers (this protocol is actually
API, but WKWebView still uses a pre-API internal protocol). This needs
to be visible from Swift, so add it to the bridging header. Unlike our
frameworks, the test bundle is not itself a module, so we do not need to
modularize UIKitSPIForTesting.h and friends to be able to use it from
swift code.

Along the way, fix a couple of public build issues, noted below.

Test: Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h

Test: Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h

* Configurations/CommonBase.xcconfig: Change the definition of
  SDKDB_HAS_148943382 to be internal only. When we generate partial
  SDKDBs from the public SDK, they do not exhibit the same bug. This
  change reverts an earlier attempt to work around the bug in
  300877@main.

* Source/WebCore/Configurations/AllowedSPI-legacy.toml: Revert an
  earlier attempt to fix the above problem (300877@main).

* Tools/TestRunnerShared/spi/UIKitSPIForTesting.h:
* Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h:

* WebKitLibraries/SDKs/iphoneos26.0-additions.sdk/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore.tbd:
  Removed. The public build gets UIKit symbols from UIKit.tbd directly,
  not by binding to UIKitCore. This sidesteps an issue where WebKitSwift
  binds to different UIKit symbols in debug and release configurations,
  which confuses extract-tbds-from-internal-sdk.
* WebKitLibraries/SDKs/iphonesimulator26.0-additions.sdk/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore.tbd: Removed.

Originally-landed-as: 300969@main (77271dc9c550). https://bugs.webkit.org/show_bug.cgi?id=300036
Canonical link: https://commits.webkit.org/301106@main
